### ISSUE TITLE
Shellcheck Inspection Suggestion and possible IntelliJ Shell Script or Shellcheck related Highlighting Issue 

### ISSUE THREAD

**Issue by mrk-han** [koalaman/shellcheck#1661]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

until ! [[ "${ADB_INSTALL_OUTPUT}" =~ "Can't find service" ]]; do
  echo ""
  ADB_INSTALL_OUTPUT=$(adb install -r "${ANDROID_APP_DESTINATION}" 2>&1)
  echo -e "\nRetrying..."
  echo ""
  echo "${ADB_INSTALL_OUTPUT}"
  sleep 2
done

```

#### Here's what shellcheck currently says:
```
From IntelliJ IDE
![image](https://user-images.githubusercontent.com/22263679/62387335-50c64080-b528-11e9-921a-9201a6f6231e.png)


From Shellcheck.net
$ shellcheck myscript
No issues detected!
```
#### Here's what I wanted or expected to see:

1) I expected it to warn me that I needed to either escape the single quote, or put the entire string into a regex like `*"Can't find service"*`
2) If you remove the single quote from `Can't` and have `=~ "Cant find service" the string does not appear green, which might mean the IDE is interpreting a regex instead of string. It would be nice to warn me that this is happening in an inspection so I can fix.

![image](https://user-images.githubusercontent.com/22263679/62387476-b1ee1400-b528-11e9-94cc-981423efb27e.png)

Bug:

- Syntax highlighting (This may be specific to the IntellJ Shell Script plugin)

New Check / Suggestion:

- Quotes are no longer needed since bash 3.2, with suggestion to fix or suggestion to use regex format

>NOTE: Quoting not necessary, as of version 3.2 of Bash.

from http://tldp.org/LDP/abs/html/bashver3.html#REGEXMATCHREF

**Comment by mrk-han** [comment]
Perhaps an inspection that  

>NOTE: Quoting not necessary, as of version 3.2 of Bash.

from http://tldp.org/LDP/abs/html/bashver3.html#REGEXMATCHREF

**Comment by koalaman** [comment]
The statement `[[ "${ADB_INSTALL_OUTPUT}" =~ "Can't find service" ]]` is correct from Bash's point of view, and ShellCheck does not emit a warning. The quotes around `"Can't find service"` are not optional, as the single quote and spaces need to be interpreted literally (For quoted patterns that probably weren't intended to match literally, there's there's already [SC2076](https://github.com/koalaman/shellcheck/wiki/SC2076)).

ShellCheck is not involved in syntax highlighting, so if IntelliJ is coloring the string in an odd way, that's an issue with something else.

**Comment by mrk-han** [comment]
Thank you @koalaman ! Do you know why SC2076 was not enabled for me? Or are there templates of checks we can download? Even when I ran shellcheck from command line this warning was not found.

Huge fan of this library, thank you so much.
